server/api/project_apis.py

```diff
@@ -440,3 +443,128 @@ def get(self, project_id: int):
             error_msg = f'Project Summary GET - unhandled error: {str(e)}'
             current_app.logger.critical(error_msg)
             return {"error": error_msg}, 500
+
+
+class TaskAnnotationsAPI(Resource):
+    def get(self, project_id: int, annotation_type: str = None):
+        """
+        Get all task annotations for a project
+        ---
+        tags:
+          - project-admin
+        produces:
+          - application/json
+        parameters:
+            - name: project_id
+              in: path
+              description: The ID of the project
+              required: true
+              type: integer
+            - name: annotation_type
+              in: path
+              description: The type of annotation to fetch
+              required: false
+              type: string
+        responses:
+            200:
+                description: Project Annotations
+            404:
+                description: Project or annotations not found
+            500:
+                description: Internal Server Error
+        """
+        try:
+            project = ProjectService.get_project_by_id(project_id)
+        except NotFound as e:
+            current_app.logger.error(f'Error validating project: {str(e)}')
+            return {"Error": "Project not found"}, 404
+
+        try:
+            if annotation_type:
+                annotations = TaskAnnotation.get_task_annotations_by_project_id_type(project_id, annotation_type)
+            else:
+                annotations = TaskAnnotation.get_task_annotations_by_project_id(project_id)
+            return annotations.to_primitive(), 200
+        except NotFound:
+            return {"Error": "Annotations not found"}, 404
+
+    def post(self, project_id: int, annotation_type: str):
+        """
+        Store new task annotations for tasks of a project
+        ---
+        tags:
+            - project-admin
+        produces:
+            - application/json
+        parameters:
+            - in: header
+              name: Content-Type
+              description: Content type for post body
+              required: true
+              type: string
+              default: application/json
+            - name: project_id
+              in: path
+              description: The unique project ID
+              required: true
+              type: integer
+              default: 1
+            - in: body
```

Comment:
I'm no good at JS, but is there a mismatch here between `body` and `annotation_str`?

Reply:
@wronk I used another endpoint as an example to write the documentation string - I think that just says it's in the body of the request - should be ok.

Reply:
I think `body` is okay for the JSON data, but we should have another item in this comment block for `annotation_type` so swagger-ui can parse it as a field to include on the api page.

Reply:
Done!